### DISCUSSION TITLE
How to persist fields when clicking "create another" ?

### DISCUSSION THREAD

**Discussion by eduardoarandah** [filamentphp/filament#9724] [Help]
I need to persist some fields when clicking "save and create another" instead of blanking the form

![image](https://github.com/filamentphp/filament/assets/4065733/fe5d6a74-3e43-4856-b395-940c1937fbf2)

I was able to do it overwriting **sendCreatedNotificationAndRedirect**

![image](https://github.com/filamentphp/filament/assets/4065733/14ebe5cc-7449-4f0c-a2b3-f158d812981c)

However this doesn't feel clean. 

Any ideas?

**Comment by leandrocfe** [comment]
Did it work?

**Reply by eduardoarandah** [reply]
yes, it works, still don't know if it's the best way to do it

**Comment by schniper** [comment]
If you used this, be careful: sendCreatedNotificationAndRedirect is no longer present.
You can use getCreatedNotification() to do stuff and just return null or the notification.

**Comment by PekkoT** [comment]
In case anyone pops in here later, this solution worked for me:
- `SettingResource` form needs to remember `category` & `client_id` between records
- The `create`-function annoyingly clears data in between and I didn't want to rewrite the entire function
- I noticed there's a `rememberData` function I could use just before record is cleared

My `CreateSetting` additions:
```
    private ?Setting $setting = null;

    public function create(bool $another = false): void
    {
        // $this->rememberData() is run inside parent::create($another)
        parent::create($another);

        if ($another && $this->setting) {
            $this->form->fill([
                'category' => $this->setting->category->value, // need the value of an enum
                'client_id' => $this->setting->client_id,
            ]);
        }
    }

    protected function rememberData(): void
    {
        $this->setting = $this->record;
        parent::rememberData();
    }
```
